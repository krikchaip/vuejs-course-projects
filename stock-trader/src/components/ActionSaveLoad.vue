<template>
  <div
    class="nav-item dropdown"
    style="cursor: pointer"
    @click="menuShow = !menuShow">
    <span class="nav-link dropdown-toggle">Save & Load</span>
    <div :class="['dropdown-menu', { show: menuShow }]">
      <div class="dropdown-item">Save Data</div>
      <div class="dropdown-item">Load Data</div>
    </div>
  </div>
</template>

<script>
  export default {
    data: () => ({
      menuShow: false
    })
  }
</script>
