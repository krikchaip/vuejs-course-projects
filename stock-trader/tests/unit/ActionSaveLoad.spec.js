import { shallowMount } from '@vue/test-utils'
import ActionSaveLoad from '@/components/ActionSaveLoad'

describe('menu is initially hidden', () => {
  const data = () => ({ menuShow: false })

  it('should toggle menu visibility when clicked', () => {
    const wrapper = shallowMount(ActionSaveLoad, { data })

    wrapper.trigger('click')
    expect(wrapper.vm.menuShow).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.trigger('click')
    expect(wrapper.vm.menuShow).toBeFalsy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
