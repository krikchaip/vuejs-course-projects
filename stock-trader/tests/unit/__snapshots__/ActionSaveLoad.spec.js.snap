// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu is initially hidden should toggle menu visibility when clicked 1`] = `
<div class="nav-item dropdown" style="cursor: pointer;"><span class="nav-link dropdown-toggle">Save &amp; Load</span>
  <div class="dropdown-menu show">
    <div class="dropdown-item">Save Data</div>
    <div class="dropdown-item">Load Data</div>
  </div>
</div>
`;

exports[`menu is initially hidden should toggle menu visibility when clicked 2`] = `
<div class="nav-item dropdown" style="cursor: pointer;"><span class="nav-link dropdown-toggle">Save &amp; Load</span>
  <div class="dropdown-menu">
    <div class="dropdown-item">Save Data</div>
    <div class="dropdown-item">Load Data</div>
  </div>
</div>
`;
